frontend/projects/ui/src/app/pages/apps-routes/app-show/app-show.page.ts

```diff
@@ -62,7 +71,9 @@ export class AppShowPage {
     private readonly navCtrl: NavController,
     private readonly patch: PatchDbService,
     private readonly connectionService: ConnectionService,
-  ) { }
+    @Inject(AbstractMarketplaceService)
```

Comment:
Is @Inject() necessary? Maybe just `private readonly marketplaceService: AbstractMarketplaceService`

Reply:
I thought it was necessary for getting the type of `getMarketplace()`, but it appears not. Will remove.

Reply:
Just kidding, it is necessary, otherwise you get this error: 

```
NullInjectorError: R3InjectorError(AppShowPageModule)[MarketplaceService -> MarketplaceService -> MarketplaceService -> MarketplaceService -> MarketplaceService]: 
  NullInjectorError: No provider for MarketplaceService!
NullInjectorError: R3InjectorError(AppShowPageModule)[MarketplaceService -> MarketplaceService -> MarketplaceService -> MarketplaceService -> MarketplaceService]: 
  NullInjectorError: No provider for MarketplaceService!
  ```

Reply:
That's a circular reference error

Reply:
Ok - this `@Inject` is used elsewhere when `MarketplaceService` is in the component constructor - is there a better pattern? Perhaps it's necessary because `MarketplaceService` extends `AbstractMarketplaceService` and this is shared?

Reply:
You can drop `@Inject` but use `AbstractMarketplaceService` as type and it will work. That is because DI works with `AbstractMarketplaceService` and we provide `MarketplaceService` as concrete implementation. Also the less concrete you define it here the better. Makes it more independent of implementation.

Reply:
Just noticed you used `getAltMarketplace` here which only exists in `MarketplaceService` implementation. Then you just leave it as is, current code is correct.